tests/components/homematicip_cloud/test_config_flow.py

```diff
@@ -76,92 +108,104 @@
+        )
 
-    result = await flow.async_step_init(user_input=config)
     assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
 
 
 async def test_import_config(hass):
     """Test importing a host with an existing config file."""
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
-
-    result = await flow.async_step_import(
-        {
-            hmipc.HMIPC_HAPID: "ABC123",
-            hmipc.HMIPC_AUTHTOKEN: "123",
-            hmipc.HMIPC_NAME: "hmip",
-        }
-    )
+    config = {
+        const.HMIPC_HAPID: "ABC123",
+        const.HMIPC_AUTHTOKEN: "123",
+        const.HMIPC_NAME: "hmip",
+    }
+
+    with patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_checkbutton",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_setup",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_register",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            const.DOMAIN, context={"source": "import"}, data=config
+        )
 
     assert result["type"] == "create_entry"
     assert result["title"] == "ABC123"
-    assert result["data"] == {
-        hmipc.HMIPC_HAPID: "ABC123",
-        hmipc.HMIPC_AUTHTOKEN: "123",
-        hmipc.HMIPC_NAME: "hmip",
-    }
+    assert result["data"] == {"authtoken": "123", "hapid": "ABC123", "name": "hmip"}
+    assert result["result"].unique_id == "ABC123"
 
 
 async def test_import_existing_config(hass):
     """Test abort of an existing accesspoint from config."""
-    flow = config_flow.HomematicipCloudFlowHandler()
-    flow.hass = hass
-
-    MockConfigEntry(
-        domain=const.DOMAIN, data={hmipc.HMIPC_HAPID: "ABC123"}
-    ).add_to_hass(hass)
-
-    result = await flow.async_step_import(
-        {
-            hmipc.HMIPC_HAPID: "ABC123",
-            hmipc.HMIPC_AUTHTOKEN: "123",
-            hmipc.HMIPC_NAME: "hmip",
-        }
-    )
+    MockConfigEntry(domain=const.DOMAIN, unique_id="ABC123").add_to_hass(hass)
+    config = {
+        const.HMIPC_HAPID: "ABC123",
+        const.HMIPC_AUTHTOKEN: "123",
+        const.HMIPC_NAME: "hmip",
+    }
+
+    with patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_checkbutton",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_setup",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.homematicip_cloud.hap.HomematicipAuth.async_register",
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            const.DOMAIN, context={"source": "import"}, data=config
+        )
 
     assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4887152)